Source/Core/VideoBackends/D3D12/D3DUtil.cpp

```diff
@@ -0,0 +1,1051 @@
+	const EfbPokeData* points,
+	size_t num_points,
+	D3D12_BLEND_DESC* blend_desc,
+	D3D12_DEPTH_STENCIL_DESC* depth_stencil_desc,
+	D3D12_VIEWPORT* viewport,
+	D3D12_CPU_DESCRIPTOR_HANDLE* render_target,
+	D3D12_CPU_DESCRIPTOR_HANDLE* depth_buffer,
+	bool rt_multisampled
+	)
+{
+	// The viewport and RT/DB are passed in so we can reconstruct the state if we need to execute in the middle of building the vertex buffer.
+
+	D3D::command_list_mgr->m_current_topology = D3D_PRIMITIVE_TOPOLOGY_TRIANGLESTRIP;
+
+	D3D12_VERTEX_BUFFER_VIEW vb_view = {
+		util_vbuf_efbpokequads->GetBuffer12()->GetGPUVirtualAddress(), // D3D12_GPU_VIRTUAL_ADDRESS BufferLocation;
+		0x100000,                                                      // UINT SizeInBytes; This is the size of the entire buffer, not just the size of the vertex data for one draw call, since the offsetting is done in the draw call itself.
+		sizeof(ColVertex)                                              // UINT StrideInBytes;
+	};
+
+	D3D::command_list_mgr->m_dirty_vertex_buffer = true;
+
+	D3D12_GRAPHICS_PIPELINE_STATE_DESC pso_desc = {
+		default_root_signature,                           // ID3D12RootSignature *pRootSignature;
+		StaticShaderCache::GetClearVertexShader(),        // D3D12_SHADER_BYTECODE VS;
+		StaticShaderCache::GetClearPixelShader(),         // D3D12_SHADER_BYTECODE PS;
+		{},                                               // D3D12_SHADER_BYTECODE DS;
+		{},                                               // D3D12_SHADER_BYTECODE HS;
+		g_ActiveConfig.iStereoMode > 0 ?
+		StaticShaderCache::GetClearGeometryShader() :
+		D3D12_SHADER_BYTECODE(),                          // D3D12_SHADER_BYTECODE GS;
+		{},                                               // D3D12_STREAM_OUTPUT_DESC StreamOutput
+		*blend_desc,                                      // D3D12_BLEND_DESC BlendState;
+		UINT_MAX,                                         // UINT SampleMask;
+		g_reset_rast_desc,                                // D3D12_RASTERIZER_DESC RasterizerState
+		*depth_stencil_desc,                              // D3D12_DEPTH_STENCIL_DESC DepthStencilState
+		StaticShaderCache::GetClearVertexShaderInputLayout(), // D3D12_INPUT_LAYOUT_DESC InputLayout
+		D3D12_INDEX_BUFFER_STRIP_CUT_VALUE_0xFFFF,        // D3D12_INDEX_BUFFER_PROPERTIES IndexBufferProperties
+		D3D12_PRIMITIVE_TOPOLOGY_TYPE_TRIANGLE,           // D3D12_PRIMITIVE_TOPOLOGY_TYPE PrimitiveTopologyType
+		1,                                                // UINT NumRenderTargets
+		{ DXGI_FORMAT_R8G8B8A8_UNORM },                   // DXGI_FORMAT RTVFormats[8]
+		DXGI_FORMAT_D24_UNORM_S8_UINT,                    // DXGI_FORMAT DSVFormat
+		{ 1 /* UINT Count */, 0 /* UINT Quality */ }      // DXGI_SAMPLE_DESC SampleDesc
+	};
+
+	if (rt_multisampled)
+	{
+		pso_desc.SampleDesc.Count = g_ActiveConfig.iMultisamples;
+	}
+
+	ID3D12PipelineState* pso = nullptr;
+	CheckHR(DX12::gx_state_cache.GetPipelineStateObjectFromCache(&pso_desc, &pso));
```

Comment:
Thanks for calling this out, there definitely isn't the need to pass blend/depth descs into the function (fixed), and so this PSO could be a simple lookup.

However, the app would have to invalidate the PSO upon stereo/msaa changes, and this is generally the only PSO lookup in D3DUtil that could be cached. The others actually use the various depth/blend states that are passed in.

The cost of this lookup is extremely cheap relative (about 0.1%) relative to the cost of the overall function, so I feel like special-casing just this particular PSO is worth it, unless this would enable a more clean way to refactor things?